commands/cat_end.py

```diff
@@ -1,37 +1,22 @@
-from db.dao.sector_dao import SectorDao
 from base.command import BaseCommand
 from base.arguments import StringArgument
-from db.base import sector_size
 from db.dao.file_dao import FileDao
+from db.dao.directory_dao import DirectoryDao
 
 
 class CatEndCommand(BaseCommand):
     command = 'cat_end'
     arguments = [StringArgument()]
 
     def run(self):
-        filename = self.arguments[0].data
-        current_directory = self.context.current_directory
+        path = self.arguments[0].data
+        file = self.context.parse(path, True)
+        file_directory = DirectoryDao.get_directory_from_directory_id(
+            file.directory_id)
 
-        if (FileDao.is_unique_filename(filename, current_directory)
-                or not FileDao.is_valid_filename(filename)):
+        if (FileDao.is_unique_filename(file.name, file_directory)
```

Comment:
Let me know if these checks are redundant and have been taken care of in the parse()